easybuild/tools/run.py

```diff
@@ -142,14 +177,23 @@ def run(cmd, fail_on_error=True, split_stderr=False, stdin=None,
         # 'input' value fed to subprocess.run must be a byte sequence
         stdin = stdin.encode()
 
+    # use bash as shell instead of the default /bin/sh used by subprocess.run
+    # (which could be dash instead of bash, like on Ubuntu, see https://wiki.ubuntu.com/DashAsBinSh)
+    if shell:
+        executable = '/bin/bash'
```

Comment:
This isn't guaranteed to exist, you should check it does or use our `which()` function

Reply:
I think it's a fair assumption that it's there, and if it's not the error message will be quite clear:

```
>>> subprocess.run("echo test", shell=True, executable='/bin/alan')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/subprocess.py", line 505, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/bin/alan'
```

Isn't that good enough, to let EasyBuild crash hard if `/bin/bash` is missing?

I don't want to check whether `/bin/bash` is *still* there every time a command is being run with `run`...
I guess I can implement a one-time check somewhere at startup, and produce a single-line error like "`ERROR: /bin/bash does not exist, EasyBuild really needs it!"`, but is that worth the effort?
In fact, if `/bin/bash` is not there, the `eb` wrapper script will fail spectacularly already, since it has `#!/bin/bash` in the shebang line:

```
bash-3.2$ eb --version
bash: eb: /bin/alan: bad interpreter: No such file or directory
```